test/integration/stats_integration_test.cc

```diff
@@ -129,5 +162,34 @@ TEST_P(StatsIntegrationTest, WithTagSpecifierWithFixedValue) {
   EXPECT_EQ(live->tags()[0].value_, "xxx");
 }
 
+INSTANTIATE_TEST_SUITE_P(IpVersions, ClusterMemoryUtilization,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// TODO(cmluciano) Refactor once envoyproxy/envoy#5624 is solved
+// TODO(cmluciano) Add options to measure multiple workers & without stats
+TEST_P(ClusterMemoryUtilization, MemoryLargeClusterSizeWithStats) {
+
+  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated() / 1000;
+  if (start_mem == 0) {
+    // Skip this test for platforms where we can't measure memory.
+    return;
+  }
+
+  auto IpVersions = TestBaseWithParam<Network::Address::IpVersion>::GetParam();
+
+  auto t1 = std::make_unique<BaseIntegrationTest>(IpVersions);
```

Comment:
Can't we just `initialize()` ourself, rather than crate a new `BaseIntegraitonTest`, since `ClusterMemoryUtilization` derives from this anyway?

Reply:
We are doing something sufficiently different here from most integration tests; it does need some code comments. In the course of this one test we are completely starting and stopping Envoy 2 or more times to measure their memory consumption under different config options, mirroring what @PiotrSikora did in his memory-measurement bash script. It was my suggestion to explicitly instantiate the test harness to get total teardown between the measurements, all within one actual test.

Reply:
Definitely needs comments then, this is a pretty uncommon pattern in the test base and flips the is-a to has-a relationship.

Reply:
Added some comments, let me know if that is clear

Reply:
Yeah, the comment makes sense, but now I'm really confused. Why do we have both a is-a and a has-a relationship? We spin up one test to figure out what's going on with 1 cluster, and then use the in-built test for multiple clusters? This seems pretty hokey. Can't we do both the 1 and 1001 test as owned integration tests?

There's also some dangerous stuff here, e.g. we delete the 1 cluster test server instance after measurement, which will free memory, then we spin up the is-a integration test but use the previously latched `m1` while `t1` was alive.

Reply:
> Yeah, the comment makes sense, but now I'm really confused. Why do we have both a is-a and a has-a relationship? We spin up one test to figure out what's going on with 1 cluster, and then use the in-built test for multiple clusters? This seems pretty hokey. Can't we do both the 1 and 1001 test as owned integration tests?

So this would be creating 2 instances of ClusterMemoryUtilization inside of this test rather than the separate call to BaseIntegration Test?

> There's also some dangerous stuff here, e.g. we delete the 1 cluster test server instance after measurement, which will free memory, then we spin up the is-a integration test but use the previously latched `m1` while `t1` was alive.

I admit this does get a bit hairy. Can we think of anything that would still allow for multiple runs within the same test case?

Reply:
Sorry, didn't realize this was blocking this PR. I think we should treat the two runs independently and uniformly; there should be no difference on whether we are running with 1 or 1001 clusters, the same methodology should be used. So, we probably need two instance of `ClusterMemoryUtilization`.